lib/internal/bootstrap_node.js

```diff
@@ -123,7 +123,7 @@
         const internalModule = NativeModule.require('internal/module');
         internalModule.addBuiltinLibsToObject(global);
         evalScript('[eval]');
-      } else if (process.argv[1]) {
+      } else if (process.argv[1] && process.argv[1] !== '-') {
```

Comment:
We already use `--` for separating arguments, and that’s what it’s commonly used for (on Unixes, at least). `-` is the most common choice for “stdin”, by far.